### ISSUE TITLE
c_windows_dbg.vsprojects/Debug/h2_http_proxy_test.exe connectivity 

### ISSUE THREAD

**Issue by kpayson64** [grpc/grpc#9088]
https://grpc-testing.appspot.com/job/gRPC_master_windows/lastCompletedBuild/testReport/(root)/c_windows_dbg/vsprojects_Debug_h2_http_proxy_test_exe_connectivity/

D1213 15:56:53.261000000 16376 test_config.c:290] test slowdown: machine=1.000000 build=1.000000 poll=1.000000 total=1.000000
I1213 15:56:53.267000000 16376 http_proxy.c:429] Proxy address: localhost:2813
D1213 15:56:53.268000000   416 connectivity.c:55] verifying
D1213 15:56:53.368000000 16376 connectivity.c:97] watching
I1213 15:56:53.370000000 13212 http_connect_handshaker.c:273] Connecting to server localhost:2811 via HTTP proxy localhost:2813
I1213 15:56:54.370000000 16376 subchannel.c:675] Connect failed: {"created":"@1481644614.370000000","description":"End of TCP stream","file":"C:\jenkins\workspace\gRPC_master_windows\workspace_c_windows_dbg\vsprojects\..\src\core\lib\iomgr\tcp_windows.c","file_line":184}
I1213 15:56:54.370000000 16376 http_connect_handshaker.c:273] Connecting to server localhost:2811 via HTTP proxy localhost:2813
I1213 15:56:54.379000000 16376 http_proxy.c:131] HTTP proxy server connect: {"created":"@1481644614.379000000","description":"OS Error","file":"C:\jenkins\workspace\gRPC_master_windows\workspace_c_windows_dbg\vsprojects\..\src\core\lib\iomgr\tcp_client_windows.c","file_line":119,"os_error":"No connection could be made because the target machine actively refused it.\r\n","syscall":"ConnectEx","wsa_error":10061}
I1213 15:56:55.372000000 11424 subchannel.c:675] Connect failed: {"created":"@1481644615.372000000","description":"End of TCP stream","file":"C:\jenkins\workspace\gRPC_master_windows\workspace_c_windows_dbg\vsprojects\..\src\core\lib\iomgr\tcp_windows.c","file_line":184}
I1213 15:56:55.372000000 11424 subchannel.c:478] Retry immediately
I1213 15:56:55.372000000 11424 subchannel.c:430] Failed to connect to channel, retrying
D1213 15:56:55.372000000 16376 connectivity.c:129] *** STARTING SERVER ***
I1213 15:56:55.372000000 11424 http_connect_handshaker.c:273] Connecting to server localhost:2811 via HTTP proxy localhost:2813
I1213 15:56:55.373000000 11424 http_proxy.c:131] HTTP proxy server connect: {"created":"@1481644615.373000000","description":"OS Error","file":"C:\jenkins\workspace\gRPC_master_windows\workspace_c_windows_dbg\vsprojects\..\src\core\lib\iomgr\tcp_client_windows.c","file_line":119,"os_error":"No connection could be made because the target machine actively refused it.\r\n","syscall":"ConnectEx","wsa_error":10061}
D1213 15:56:55.380000000 16376 connectivity.c:134] *** STARTED SERVER ***
D1213 15:56:55.380000000 16376 connectivity.c:151] *** SHUTTING DOWN SERVER ***
E1213 15:56:58.380000000 16376 cq_verifier.c:184] assertion failed: e->success == ev->success
Abort handler called.
*** 0: 00000000002E481F  - 0000000000000000
*** 1: 00000000002E4D27  - 0000000000000000
*** 2: 0000000000386619  - 0000000000000000
*** 3: 0000000000381A14  - 0000000000000000
*** 4: 000000000054CB0D  - 0000000000000000
*** 5: 000000000054C0B1  - 0000000000000000
*** 6: 000000000035D680  - 0000000000000000
*** 7: 000000000035D83F  - 0000000000000000
*** 8: 000000000034FEBB  - 0000000000000000
*** 9: 00000000002E13AA  - 0000000000000000
*** 10: 0000000000381D7A  - 0000000000000000
*** 11: 0000000000381ECD  - 0000000000000000
*** 12: 000000007534336A BaseThreadInitThunk - 0000000075343358
*** 13: 0000000077279902 RtlInitializeExceptionChain - 000000007727989F
*** 14: 00000000772798D5 RtlInitializeExceptionChain - 000000007727989F

**Comment by nathanielmanistaatgoogle** [comment]
Seen in `gRPC_master_windows` builds:

* [1117](https://grpc-testing.appspot.com/job/gRPC_master_windows/1117/testReport/junit/(root)/c_windows_opt/vsprojects_Release_h2_http_proxy_test_exe_connectivity/)
* [1124](https://grpc-testing.appspot.com/job/gRPC_master_windows/1124/testReport/junit/(root)/c_windows_opt/vsprojects_Release_h2_http_proxy_test_exe_connectivity/)
* [1129](https://grpc-testing.appspot.com/job/gRPC_master_windows/1129/testReport/junit/(root)/c_windows_opt/vsprojects_Release_h2_http_proxy_test_exe_connectivity/)

**Comment by y-zeng** [comment]
Seen again:
https://grpc-testing.appspot.com/job/gRPC_pr_win/1974/

**Comment by dgquintas** [comment]
Seen again on Master:

https://grpc-testing.appspot.com/job/gRPC_master_windows/1706/testReport/junit/(root)/c_windows_dbg/vsprojects_Debug_h2_http_proxy_test_exe_connectivity/

```
D0224 15:59:03.008000000 13136 test_config.c:379] test slowdown factor: sanitizer=1, fixture=1, poller=1, total=1
I0224 15:59:03.014000000 13136 http_proxy.c:440] Proxy address: localhost:2914
D0224 15:59:03.015000000 12448 connectivity.c:55] verifying
D0224 15:59:03.115000000 13136 connectivity.c:97] watching
I0224 15:59:03.117000000  8796 http_connect_handshaker.c:319] Connecting to server localhost:2911 via HTTP proxy ipv4:127.0.0.1:2914
I0224 15:59:04.117000000  8796 subchannel.c:706] Connect failed: {"created":"@1487951944.117000000","description":"TCP stream shutting down","file":"C:\jenkins\workspace\gRPC_master_windows\workspace_c_windows_dbg\vsprojects\..\src\core\lib\iomgr\tcp_windows.c","file_line":189,"referenced_errors":[{"created":"@1487951944.117000000","description":"Handshake timed out","file":"C:\jenkins\workspace\gRPC_master_windows\workspace_c_windows_dbg\vsprojects\..\src\core\lib\channel\handshaker.c","file_line":241}]}
I0224 15:59:04.118000000  8796 http_proxy.c:134] HTTP proxy server connect: {"created":"@1487951944.118000000","description":"OS Error","file":"C:\jenkins\workspace\gRPC_master_windows\workspace_c_windows_dbg\vsprojects\..\src\core\lib\iomgr\tcp_client_windows.c","file_line":120,"os_error":"No connection could be made because the target machine actively refused it.\r\n","syscall":"ConnectEx","wsa_error":10061}
I0224 15:59:04.119000000  8796 http_connect_handshaker.c:319] Connecting to server localhost:2911 via HTTP proxy ipv6:[::1]:2914
I0224 15:59:05.113000000  8796 http_proxy.c:134] HTTP proxy server connect: {"created":"@1487951945.113000000","description":"OS Error","file":"C:\jenkins\workspace\gRPC_master_windows\workspace_c_windows_dbg\vsprojects\..\src\core\lib\iomgr\tcp_client_windows.c","file_line":120,"os_error":"No connection could be made because the target machine actively refused it.\r\n","syscall":"ConnectEx","wsa_error":10061}
I0224 15:59:05.114000000  8796 subchannel.c:706] Connect failed: {"created":"@1487951945.114000000","description":"End of TCP stream","file":"C:\jenkins\workspace\gRPC_master_windows\workspace_c_windows_dbg\vsprojects\..\src\core\lib\iomgr\tcp_windows.c","file_line":190}
I0224 15:59:05.114000000  8796 subchannel.c:502] Retry immediately
I0224 15:59:05.114000000  8796 subchannel.c:454] Failed to connect to channel, retrying
D0224 15:59:05.115000000 13136 connectivity.c:129] *** STARTING SERVER ***
I0224 15:59:05.115000000  8796 http_connect_handshaker.c:319] Connecting to server localhost:2911 via HTTP proxy ipv4:127.0.0.1:2914
D0224 15:59:05.125000000 13136 connectivity.c:134] *** STARTED SERVER ***
D0224 15:59:05.125000000 13136 connectivity.c:151] *** SHUTTING DOWN SERVER ***
E0224 15:59:08.126000000 13136 cq_verifier.c:200] assertion failed: e->success == ev->success
Abort handler called.
*** 0: 0000000000EF4C5F  - 0000000000000000
*** 1: 0000000000EF5167  - 0000000000000000
*** 2: 0000000000FA2EE9  - 0000000000000000
*** 3: 0000000000F9E074  - 0000000000000000
*** 4: 000000000116AEAD  - 0000000000000000
*** 5: 000000000116A344  - 0000000000000000
*** 6: 0000000000F7601E  - 0000000000000000
*** 7: 0000000000F761DF  - 0000000000000000
*** 8: 0000000000F68DCF  - 0000000000000000
*** 9: 0000000000EF13AA  - 0000000000000000
*** 10: 0000000000F9E3DA  - 0000000000000000
*** 11: 0000000000F9E52D  - 0000000000000000
*** 12: 00000000772D336A BaseThreadInitThunk - 00000000772D3358
*** 13: 0000000077D89902 RtlInitializeExceptionChain - 0000000077D8989F
*** 14: 0000000077D898D5 RtlInitializeExceptionChain - 0000000077D8989F
```

It does look like a timeout.

**Comment by apolcyn** [comment]
the same error seen again on multiple recent master windows runs:

log in https://grpc-testing.appspot.com/job/gRPC_master_windows/2366/#showFailuresLink

```
D0424 22:13:36.501000000  3196 connectivity.c:129] *** STARTING SERVER ***
I0424 22:13:36.501000000 11644 http_connect_handshaker.c:319] Connecting to server localhost:1710 via HTTP proxy ipv4:127.0.0.1:1751
D0424 22:13:36.502000000 11644 http_proxy_fixture.c:328] on_read_request_done: 0047AE90 "No Error"
D0424 22:13:36.504000000  3196 connectivity.c:134] *** STARTED SERVER ***
D0424 22:13:36.504000000  3196 connectivity.c:151] *** SHUTTING DOWN SERVER ***
I0424 22:13:36.523000000  3196 http_proxy_fixture.c:145] HTTP proxy server connect: {"created":"@1493072016.523000000","description":"OS Error","file":"C:\jenkins\workspace\gRPC_master_windows\workspace_c_windows_opt_native\vsprojects\..\src\core\lib\iomgr\tcp_client_windows.c","file_line":120,"os_error":"No connection could be made because the target machine actively refused it.\r\n","syscall":"ConnectEx","wsa_error":10061}
D0424 22:13:36.523000000  3196 http_proxy_fixture.c:117] endpoints: 0046A270 00000000
E0424 22:13:39.506000000  3196 cq_verifier.c:200] assertion failed: e->success == ev->success
Abort handler called.
```

**Comment by ericgribkoff** [comment]
Seen again: https://grpc-testing.appspot.com/job/gRPC_master_windows/2421/

**Comment by jtattermusch** [comment]
```
D0509 14:44:06.447000000  2544 test_config.c:403] test slowdown factor: sanitizer=1, fixture=1, poller=1, total=1
D0509 14:44:06.449000000  2544 dns_resolver.c:316] Using native dns resolver
I0509 14:44:06.462000000  2544 http_proxy_fixture.c:459] Proxy address: localhost:1208
D0509 14:44:06.463000000 13708 connectivity.c:55] verifying
D0509 14:44:06.562000000  2544 connectivity.c:97] watching
I0509 14:44:06.590000000 14292 http_connect_handshaker.c:319] Connecting to server localhost:1203 via HTTP proxy ipv4:127.0.0.1:1208
D0509 14:44:06.590000000 14292 http_proxy_fixture.c:328] on_read_request_done: 00638790 "No Error"
I0509 14:44:07.587000000 14292 subchannel.c:707] Connect failed: {"created":"@1494341047.587000000","description":"TCP stream shutting down","file":"C:\jenkins\workspace\gRPC_pr_win\workspace_c_windows_dbg_native\vsprojects\..\src\core\lib\iomgr\tcp_windows.c","file_line":189,"referenced_errors":[{"created":"@1494341047.587000000","description":"Handshake timed out","file":"C:\jenkins\workspace\gRPC_pr_win\workspace_c_windows_dbg_native\vsprojects\..\src\core\lib\channel\handshaker.c","file_line":241}]}
I0509 14:44:07.587000000 14292 http_connect_handshaker.c:319] Connecting to server localhost:1203 via HTTP proxy ipv6:[::1]:1208
D0509 14:44:07.587000000 14292 http_proxy_fixture.c:328] on_read_request_done: 00637118 "No Error"
I0509 14:44:07.598000000 14292 http_proxy_fixture.c:145] HTTP proxy server connect: {"created":"@1494341047.598000000","description":"OS Error","file":"C:\jenkins\workspace\gRPC_pr_win\workspace_c_windows_dbg_native\vsprojects\..\src\core\lib\iomgr\tcp_client_windows.c","file_line":120,"os_error":"No connection could be made because the target machine actively refused it.\r\n","syscall":"ConnectEx","wsa_error":10061}
D0509 14:44:07.598000000 14292 http_proxy_fixture.c:117] endpoints: 00631518 00000000
I0509 14:44:08.580000000 14292 http_proxy_fixture.c:145] HTTP proxy server connect: {"created":"@1494341048.580000000","description":"OS Error","file":"C:\jenkins\workspace\gRPC_pr_win\workspace_c_windows_dbg_native\vsprojects\..\src\core\lib\iomgr\tcp_client_windows.c","file_line":120,"os_error":"No connection could be made because the target machine actively refused it.\r\n","syscall":"ConnectEx","wsa_error":10061}
D0509 14:44:08.580000000 14292 http_proxy_fixture.c:117] endpoints: 00636F50 00000000
I0509 14:44:08.580000000 14292 subchannel.c:707] Connect failed: {"created":"@1494341048.580000000","description":"End of TCP stream","file":"C:\jenkins\workspace\gRPC_pr_win\workspace_c_windows_dbg_native\vsprojects\..\src\core\lib\iomgr\tcp_windows.c","file_line":190}
I0509 14:44:08.580000000 14292 subchannel.c:500] Retry immediately
I0509 14:44:08.580000000 14292 subchannel.c:452] Failed to connect to channel, retrying
D0509 14:44:08.580000000  2544 connectivity.c:129] *** STARTING SERVER ***
I0509 14:44:08.580000000 14292 http_connect_handshaker.c:319] Connecting to server localhost:1203 via HTTP proxy ipv4:127.0.0.1:1208
D0509 14:44:08.581000000 14292 http_proxy_fixture.c:328] on_read_request_done: 00638088 "No Error"
D0509 14:44:08.588000000  2544 connectivity.c:134] *** STARTED SERVER ***
D0509 14:44:08.588000000  2544 connectivity.c:151] *** SHUTTING DOWN SERVER ***
E0509 14:44:11.591000000  2544 cq_verifier.c:244] actual success does not match expected: 00000005 GRPC_OP_COMPLETE success=1 C:\jenkins\workspace\gRPC_pr_win\workspace_c_windows_dbg_native\vsprojects\..\test\core\end2end\tests\connectivity.c:159
Abort handler called.
*** 0: 0000000001194D2F  - 0000000000000000
*** 1: 0000000001195237  - 0000000000000000
*** 2: 0000000001255F29  - 0000000000000000
*** 3: 0000000001251084  - 0000000000000000
*** 4: 000000000141E4DE  - 0000000000000000
*** 5: 000000000141D6D4  - 0000000000000000
*** 6: 0000000001223C54  - 0000000000000000
*** 7: 0000000001223E1F  - 0000000000000000
*** 8: 0000000001215857  - 0000000000000000
*** 9: 00000000011913AA  - 0000000000000000
*** 10: 00000000012513EA  - 0000000000000000
*** 11: 000000000125153D  - 0000000000000000
*** 12: 0000000074E1336A BaseThreadInitThunk - 0000000074E13358
*** 13: 0000000077389902 RtlInitializeExceptionChain - 000000007738989F
*** 14: 00000000773898D5 RtlInitializeExceptionChain - 000000007738989F
```

**Comment by makdharma** [comment]
Should be fixed with #11224
